### DISCUSSION TITLE
Setting shadows using GLTFModel

### DISCUSSION THREAD

**Discussion by greegus** [Tresjs/tres#484] [Q&A] [answered]
I'm trying to set shadow casting from a GLTF model imported using `GLTFModel` component. I wasn't able to find it mentioned in the docs, so I just naively added the `cast-shadows` attr.

```vue
<TresCanvas shadows>
  <TresDirectionalLight :position="[5, 10, 5]" :direction="[1, -2, 0]" cast-shadows />
  <GLTFModel path="path/to/model.gltf" cast-shadows />
  <Backdrop receive-shadow />
</TresCanvas>
```

This, however, just produce the following error:

```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'shadows')
    at tres.js:560:32
    at Array.reduce (<anonymous>)
    at patchProp (tres.js:560:15)
    at mountElement (runtime-core.esm-bundler.js:5293:11)
    at processElement (runtime-core.esm-bundler.js:5240:7)
    at patch (runtime-core.esm-bundler.js:5112:11)
    at ReactiveEffect.componentUpdateFn [as fn] (runtime-core.esm-bundler.js:5792:11)
    at ReactiveEffect.run (reactivity.esm-bundler.js:178:19)
    at instance.update (runtime-core.esm-bundler.js:5898:51)
    at setupRenderEffect (runtime-core.esm-bundler.js:5906:5)
```

These are the dependency version in the project

```
"@tresjs/cientos": "^3.7.0",
"@tresjs/core": "^3.6.0",
"three": "^0.160.0",
```

What is the corrent way to set up shadow casting (and receiving shadow) by a model imported this way? Thanks! 🙏

**Comment by JaimeTorrealba** [comment, marked as answer]
Right now we have to add the reference, Then use the (Traverse)[https://threejs.org/docs/index.html?q=object#api/en/core/Object3D.traverse] function to tell all the meshes to add cast-shadow.
Something like:

<script setup>
const myModelRef = shallowRef()

watch(myModelRef , value => {
console.log(value) //here you have to see the structure of your model to apply traverse
})
</script>
 <TresCanvas shadows>
  <TresDirectionalLight :position="[5, 10, 5]" :direction="[1, -2, 0]" cast-shadows />
  <GLTFModel path="path/to/model.gltf" ref="myModelRef" />
  <Backdrop receive-shadow />
</TresCanvas>

But this seems like a big opportunity to improve how the GLTFModel  works, we could add a issue on cientos if you agree

**Comment by alvarosabu** [comment]
@JaimeTorrealba I agree is a great opportunity, could you please create a ticket for this feature?

**Comment by JaimeTorrealba** [comment]
@greegus soon in Cientos https://github.com/Tresjs/cientos/pull/330
Thanks for the Idea

**Reply by greegus** [reply]
Awesome! Thanks for looking into that!
